exchanges/exchange.go

```diff
@@ -945,3 +958,110 @@ func (e *Base) FormatWithdrawPermissions() string {
 
 	return NoAPIWithdrawalMethodsText
 }
+
+// GetOrdersRequest used for GetOrderHistory and GetOpenOrders wrapper functions
+type GetOrdersRequest struct {
+	OrderType  OrderType
+	OrderSide  OrderSide
+	StartTicks time.Time
+	EndTicks   time.Time
+	// Currencies Empty array = all currencies. Some endpoints only support singular currency enquiries
+	Currencies []pair.CurrencyPair
+}
+
+// OrderStatus defines order status types
+type OrderStatus string
+
+// All OrderStatus types
+const (
+	AnyOrderStatus             OrderStatus = "ANY"
+	NewOrderStatus             OrderStatus = "NEW"
+	ActiveOrderStatus          OrderStatus = "ACTIVE"
+	PartiallyFilledOrderStatus OrderStatus = "PARTIALLY_FILLED"
+	FilledOrderStatus          OrderStatus = "FILLED"
+	CancelledOrderStatus       OrderStatus = "CANCELED"
+	PendingCancelOrderStatus   OrderStatus = "PENDING_CANCEL"
+	RejectedOrderStatus        OrderStatus = "REJECTED"
+	ExpiredOrderStatus         OrderStatus = "EXPIRED"
+	HiddenOrderStatus          OrderStatus = "HIDDEN"
+	UnknownOrderStatus         OrderStatus = "UNKNOWN"
+)
+
+// FilterOrdersBySide removes any OrderDetails that don't match the orderStatus provided
+func (e *Base) FilterOrdersBySide(orders *[]OrderDetail, orderSide OrderSide) {
```

Comment:
Better performance can be achieved by using go's built in sort package. Please see this as a reference: 

https://github.com/thrasher-/gocryptotrader/blob/master/exchanges/stats/stats.go#L24